### PR TITLE
fix: rename superblock to use /2022/

### PR DESCRIPTION
**Pull request by ojeytonwilliams**
I was going to just let us use the `superBlock` property (rather than the directory), but once I'd done that, I figured I might as well rename the slug while I'm at it.

As for the directory name: 14-responsive-web-design-22, we don't need the 22 any more.  I didn't rename that here, but it's just a organisational convenience now, so it can be whatever we like.

I've not done anything about the blocks, just superblocks.  But I think this is a step in the right direction.

### CODE REVIEW

File: cypress/integration/learn/challenges/projects.js [hunk 1]
```diff
@@ -64,7 +64,7 @@ describe('project submission', () => {
     () => {
       cy.fixture('../../config/curriculum.json').then(curriculum => {
         const { challenges, meta } =
-          curriculum[SuperBlocks.JsAlgoDataStruct].blocks[
+          curriculum['02-javascript-algorithms-and-data-structures'].blocks[
```
Comment (nhcarrigan) [inline cypress/integration/learn/challenges/projects.js:67]: Why can't the enum be used anymore?

Comment (ShaunSHamilton) [reply cypress/integration/learn/challenges/projects.js:67]: The real question is why does the `02-` need to be included now?

Comment (nhcarrigan) [reply cypress/integration/learn/challenges/projects.js:67]: It's part of my secret mission to put the number everywhere. While you're distracted by the numbers, I'll start sneaking in jQuery. 😁

**Comment by nhcarrigan** [comment]
So what happens in 2023?

File: cypress/integration/learn/challenges/projects.js (hunk 1 above)
Comment (ojeytonwilliams) [reply cypress/integration/learn/challenges/projects.js:67]: So, this is the one place (that I can find...) where we index the curriculum, other than during the build.  I've changed the build to stop stripping the `02-` etc off and instead just build an object like this

```js
var curriculum = {
"00-certifications": { blocks: [...] },
"01-responsive-web-design": { blocks: [...] },
"02-javascript-algorithms-and-data-structures": { blocks: [...] }
}
```

but that's only used in two places: here and during the build itself.  Everywhere else just wants a set of challenges and doesn't care how they're organised.

That is to say, this is a build artifact that we're abusing during testing, we could search for those challenges if we want.  Might be better, tbh, so if you want I'll refactor it.

**Comment by ojeytonwilliams** [comment]
> So what happens in 2023?

That we can decide tomorrow :)  If you all want to get this in without the rename, that's totally cool.  Just revert 8d4450df19c20d520c7ac7a6ff153bb5f8899c10 and it should be fine.

File: cypress/integration/learn/challenges/projects.js (hunk 1 above)
Comment (ShaunSHamilton) [reply cypress/integration/learn/challenges/projects.js:67]: Erm... I would still like these magical numbers to disappear everywhere.

Comment (ojeytonwilliams) [reply cypress/integration/learn/challenges/projects.js:67]: Right. This decouples the dashed name of the superblock from the dir, so we're free to use more sensible names.

I don't have a strong opinion what names we should use, but this PR gives us that ability.

**Review by naomi-lgbt — APPROVED** [review]
I think I'm happy with this.

**Review by moT01 — APPROVED** [review]
This LGTM 🎉

**Pull request merged**